.github/workflows/ci.yml

```diff
@@ -16,11 +17,42 @@ env:
   GO_VERSION: '1.21'
 
 jobs:
+  ci-target-check:
+    runs-on: ubuntu-latest
+
+    outputs:
+      build: ${{ steps.ci-target-check.outputs.build }}
+      bench: ${{ steps.ci-target-check.outputs.bench }}
+      sharding-test: ${{ steps.ci-target-check.outputs.sharding-test }}
+
+    steps:
+    - name: CI target check by path
+      uses: dorny/paths-filter@v3
+      id: ci-target-check
+      with:
+        filters: |
+          build: '**'
+          bench: 
```

Comment:
How about simplifying below packages like this?

```go
            - 'pkg/**'
            - 'server/**'
            - 'client/**'
            - 'admin/**'
            - 'api/converter/**'
```

For `api/types/**` and `/test/helper/**`, we can skip it because it has low impact on the bench.
Also simplifying to `pkg/**` because this package is all about CRDT itself.

Reply:
Thank you for your quick review!

I was a bit concerned that I might have defined the bench target too narrowly, 
but I think your suggestion is much better!